base_jsonify/models/ir_export.py

```diff
@@ -23,41 +39,76 @@ def update_dict(data, fields):
     if len(fields) == 1:
         if field == ".id":
             field = "id"
-        data[field] = True
+        data[field] = (True, options)
     else:
         if field not in data:
-            data[field] = OrderedDict()
-        update_dict(data[field], fields[1:])
+            data[field] = (False, OrderedDict())
+        update_dict(data[field][1], fields[1:], options)
 
 
 def convert_dict(dict_parser):
     """Convert dict returned by update_dict to list consistent w/ Odoo API.
 
-    The list is composed of strings (field names or aliases) or tuples.
+    The list is composed of strings (field names or targets) or tuples.
     """
     parser = []
     for field, value in dict_parser.items():
-        if value is True:
-            parser.append(field)
+        if value[0] is True:  # is a leaf
+            parser.append(field_dict(field, value[1]))
         else:
-            parser.append((field, convert_dict(value)))
+            parser.append((field_dict(field), convert_dict(value[1])))
     return parser
 
 
+def field_dict(field, options=None):
+    result = {"name": field.split(":")[0]}
+    if len(field.split(":")) > 1:
+        result["target"] = field.split(":")[1]
+    for option in options or {}:
+        if options[option]:
+            result[option] = options[option]
+    return result
+
+
 class IrExport(models.Model):
     _inherit = "ir.exports"
 
+    language_agnostic = fields.Boolean(
+        default=False,
+        string="Language Agnostic",
+        help="If set, will set the lang to False when exporting lines without lang,"
+        " otherwise it uses the lang in the given context to export these fields",
+    )
+
+    global_resolver_id = fields.Many2one(
+        comodel_name="ir.exports.resolver",
+        string="Custom global resolver",
+        domain="[('type', '=', 'global')]",
+        help="If set, will apply the global resolver to the result",
+    )
+
     def get_json_parser(self):
```

Comment:
```suggestion
    @tools.ormcache("self.language_agnostic", "self.global_resolver_id.id", "self.export_fields.ids")
    def get_json_parser(self):
```
to be adapted in case there are other keys to consider

Reply:
`@ormcache("self.language_agnostic", "self.global_resolver_id.id", "tuple(self.export_fields.ids)")`
 
This is the only comment that I have not implemented.
Because it depends on `self.export_fields.ids`, if you actually modify one line, the returned parser is not updated (the One2many itself hasn't changed).

You can check for yourself, as the second test in test_getting_parser breaks as soon as you put the decorator.

Reply:
> Because it depends on self.export_fields.ids, if you actually modify one line, the returned parser is not updated (the One2many itself hasn't changed).

what about defining a custom cache key using `self.export_fields.mapped("write_date")`?